modules/apps/portal/portal-company-log/src/main/java/com/liferay/portal/company/log/internal/servlet/CompanyLogServlet.java

```diff
@@ -199,21 +198,12 @@ private void _downloadLogFile(
 				--start;
 			}
 
-			byte[] bytes = new byte[(int)logFile.length()];
-
-			try (UnsyncBufferedInputStream unsyncBufferedInputStream =
-					new UnsyncBufferedInputStream(
-						Files.newInputStream(logFile.toPath()));
-				 UnsyncByteArrayOutputStream unsyncByteArrayOutputStream =
-					new UnsyncByteArrayOutputStream()) {
-
-				unsyncBufferedInputStream.read(bytes);
-
-				unsyncByteArrayOutputStream.write(bytes, start, end - start);
+			try (FileChannel fileChannel = FileChannel.open(logFile.toPath())) {
+				fileChannel.position(start);
 
 				ServletResponseUtil.sendFile(
 					httpServletRequest, httpServletResponse, logFile.getName(),
-					unsyncByteArrayOutputStream.toByteArray(),
+					Channels.newInputStream(fileChannel), end - start,
```

Comment:
@dantewang, I saw your changes when I was ready to write new changes (I didn't careful check my email). I had one rule  by using Channels.newInputStream() as inputStream, but I didn't find how to make the param as ReadableByteChannel. Learn it from your changes. Thanks.